svc/pkg/cluster/worker/src/workers/datacenter_tls_issue.rs

```diff
@@ -0,0 +1,281 @@
+async fn worker(
+	ctx: &OperationContext<cluster::msg::datacenter_tls_issue::Message>,
+) -> GlobalResult<()> {
+	let datacenter_id = unwrap_ref!(ctx.datacenter_id).as_uuid();
+
+	// Create CF client
+	let cf_token = util::env::read_secret(&["cloudflare", "terraform", "auth_token"]).await?;
+	let client = cf_framework::async_api::Client::new(
+		cf_framework::auth::Credentials::UserAuthToken { token: cf_token },
+		Default::default(),
+		cf_framework::Environment::Production,
+	)
+	.map_err(crate::CloudflareError::from)?;
+
+	// Fetch ACME account registration
+	let account = acme_account().await?;
+
+	let base_zone_id = unwrap!(
+		util::env::cloudflare::zone::base::id(),
+		"dns not configured"
+	);
+	let job_zone_id = unwrap!(util::env::cloudflare::zone::job::id(), "dns not configured");
+	let domain_main = unwrap!(util::env::domain_main(), "dns not enabled");
+	let domain_job = unwrap!(util::env::domain_job(), "dns not enabled");
+
+	// NOTE: We don't use try_join because these run in parallel, the dns record needs to be deleted for each
+	// order upon failure
+	let (gg_cert, job_cert) = tokio::join!(
+		order(
+			&client,
+			ctx.renew,
+			base_zone_id,
+			&account,
+			domain_main,
+			vec![format!("*.{datacenter_id}.{domain_main}")],
+		),
+		order(
+			&client,
+			ctx.renew,
+			job_zone_id,
+			&account,
+			domain_job,
+			vec![
+				format!("*.lobby.{datacenter_id}.{domain_job}"),
+				format!("*.{datacenter_id}.{domain_job}"),
+			],
+		),
+	);
+	let (gg_cert, job_cert) = (gg_cert?, job_cert?);
+
+	sql_execute!(
+		[ctx]
+		"
+		UPDATE db_cluster.datacenter_tls
+		SET
+			gg_cert_pem = $2,
+			gg_private_key_pem = $3,
+			job_cert_pem = $4,
+			job_private_key_pem = $5,
+			state = $6,
+			expire_ts = $7
+		WHERE datacenter_id = $1
+		",
+		datacenter_id,
+		gg_cert.certificate(),
+		gg_cert.private_key(),
+		job_cert.certificate(),
+		job_cert.private_key(),
+		backend::cluster::TlsState::Active as i64,
+		util::timestamp::now() + util::duration::days(gg_cert.valid_days_left()),
+	)
+	.await?;
+
+	Ok(())
+}
+
+async fn acme_account() -> GlobalResult<Account<MemoryPersist>> {
+	// TODO: Use DirectoryUrl::LetsEncryptStaging for dev
+	let url = DirectoryUrl::LetsEncrypt;
+	let persist = MemoryPersist::new();
+
+	// Write account private key (from terraform) to persistence
+	let pem_key = PersistKey::new(
+		ENCRYPT_EMAIL,
+		PersistKind::AccountPrivateKey,
+		"acme_account",
+	);
+	let pem = util::env::var("TLS_ACME_ACCOUNT_PRIVATE_KEY_PEM")?;
+	persist.put(&pem_key, pem.as_bytes())?;
+
+	// Get ACME account info
+	let acc = tokio::task::spawn_blocking(move || {
+		// Initialize ACME directory
+		let dir = Directory::from_url(persist, url)?;
+
+		tracing::info!("fetching account");
+		dir.account(ENCRYPT_EMAIL)
+	})
+	.await??;
+
+	Ok(acc)
+}
+
+// TODO: This function contains both blocking calls that cannot be shared between threads and async calls.
+// Maybe theres a way to defer the blocking calls somehow
```

Comment:
Yep, this needs to be fixed. If not, it will block _all_ workers and cause massive pauses on the worker. Does `block_in_place` work?